diff --git a/docroot/sites/all/modules/contrib/legal/legal.module b/docroot/sites/all/modules/contrib/legal/legal.module
index 5abb81c..39e62c8 100644
--- a/docroot/sites/all/modules/contrib/legal/legal.module
+++ b/docroot/sites/all/modules/contrib/legal/legal.module
@@ -700,7 +700,7 @@ function legal_get_conditions($language = NULL) {
 
   if (!empty($language)) {
     $result = db_select('legal_conditions', 'lc')
-      ->fields('lc')
+      ->fields('lc', $keys)
       ->condition('language', $language)
       ->orderBy('version', 'DESC')
       ->orderBy('revision', 'DESC')
@@ -711,7 +711,7 @@ function legal_get_conditions($language = NULL) {
   }
   else {
     $result = db_select('legal_conditions', 'lc')
-      ->fields('lc')
+      ->fields('lc', $keys)
       ->orderBy('tc_id', 'DESC')
       ->execute()
       ->fetchAllAssoc('tc_id');
