diff --git a/modules/file/file.module b/modules/file/file.module
index 3d351fa..b4a78fe 100644
--- a/modules/file/file.module
+++ b/modules/file/file.module
@@ -242,7 +242,7 @@ function file_ajax_upload() {
     // Invalid request.
     drupal_set_message(t('An unrecoverable error occurred. The uploaded file likely exceeded the maximum file size (@size) that this server supports.', array('@size' => format_size(file_upload_max_size()))), 'error');
     $commands = array();
-    $commands[] = ajax_command_replace(NULL, theme('status_messages'));
+    $commands[] = ajax_command_prepend(NULL, theme('status_messages'));
     return array('#type' => 'ajax', '#commands' => $commands);
   }
 
