diff --git a/label_help.module b/label_help.module
index 0859851..d700c02 100644
--- a/label_help.module
+++ b/label_help.module
@@ -15,43 +15,53 @@
  * Implements hook_form_alter().
  */
 function label_help_form_alter(&$form, &$form_state, $form_id) {
-  foreach ($form as $key => $item) {
-    if (is_array($item) && isset($item[LANGUAGE_NONE])) {
-      // Put comments above the label for field forms of type 'container'
-      // that are specifically configured.
-      if (isset($item['#type']) && $item['#type'] == 'container' && isset($item[LANGUAGE_NONE][0])
-        && isset($item[LANGUAGE_NONE][0]['#entity_type']) && isset($item[LANGUAGE_NONE][0]['#field_name']) && isset($item[LANGUAGE_NONE][0]['#bundle'])
-      ) {
-        $entity_type = $item[LANGUAGE_NONE][0]['#entity_type'];
-        $field_name = $item[LANGUAGE_NONE][0]['#field_name'];
-        $bundle_name = $item[LANGUAGE_NONE][0]['#bundle'];
-        $instance = field_info_instance($entity_type, $field_name, $bundle_name);
-        if (isset($instance['widget']['settings']['label_help_description'])) {
-          // For reasons best known to the lunatics who designed the Forms API,
-          // $form[$key][LANGUAGE_NONE][0]['#theme_options'] has to be set to get
-          // this working for textarea fields, and
-          // form[$key][LANGUAGE_NONE][0]['value']['#theme_options' has to be set
-          // to get this working for one-line text fields.
-          $form[$key][LANGUAGE_NONE][0]['#theme_options'] = $form[$key][LANGUAGE_NONE][0]['value']['#theme_options'] = array(
-            'description at top' => $instance['widget']['settings']['label_help_description'],
-          );
-        }
+  $form = label_help_set_description($form);
+}
+
+/**
+ * Recursively loops through form elements and sets the appropriate label_help description.
+ * 
+ * @param  [type] $form [description]
+ * @return [type]       [description]
+ */
+function label_help_set_description($form) {
+  foreach ($form as $key => &$field) {
+    if ($key{0} != '#' && is_array($field)) {
+      $lang = isset($field['#language']) ? $field['#language'] : LANGUAGE_NONE;
+
+      if (!isset($field[$lang][0]['#entity_type']) || !isset($field[$lang][0]['#field_name']) || !isset($field[$lang][0]['#bundle'])) {
+        continue;
       }
-      // Move comments to the top for other field forms that
-      // are specifically configured.
-      elseif (isset($item[LANGUAGE_NONE]['#entity_type']) && isset($item[LANGUAGE_NONE]['#field_name']) && isset($item[LANGUAGE_NONE]['#bundle'])) {
-        $entity_type = $item[LANGUAGE_NONE]['#entity_type'];
-        $field_name = $item[LANGUAGE_NONE]['#field_name'];
-        $bundle_name = $item[LANGUAGE_NONE]['#bundle'];
-        $instance = field_info_instance($entity_type, $field_name, $bundle_name);
-        if (isset($instance['widget']['settings']['label_help_description'])) {
-          $form[$key][LANGUAGE_NONE]['#theme_options'] = $form[$key][LANGUAGE_NONE]['value']['#theme_options'] = array(
-            'description at top' => $instance['widget']['settings']['label_help_description'],
-          );
+
+      $entity_type = $field[$lang][0]['#entity_type'];
+      $field_name = $field[$lang][0]['#field_name'];
+      $bundle_name = $field[$lang][0]['#bundle'];
+
+      if (isset($field['#type']) && $field['#type'] == 'container' && isset($form['#entity_type']) && isset($form['#bundle'])) {
+        $entity_type = $form['#entity_type'];
+        $bundle_name = $form['#bundle'];
+      }
+
+      $instance = field_info_instance($entity_type, $field_name, $bundle_name);
+
+      if (isset($instance['widget']['settings']['label_help_description'])) {
+        $field[$lang][0]['#theme_options']['description at top']          = $instance['widget']['settings']['label_help_description'];
+        $field[$lang][0]['value']['#theme_options']['description at top'] = $instance['widget']['settings']['label_help_description'];
+      }
+
+      if (isset($field[$lang])) {
+        foreach ($field[$lang] as &$child) {
+          if (is_array($child)) {
+            $child = label_help_set_description($child);
+          }
         }
       }
+
+      $form[$key] = $field;
     }
   }
+
+  return $form;
 }
 
 /**
