diff --git a/sites/all/modules/contrib/legal/legal.module b/sites/all/modules/contrib/legal/legal.module
index 092fce6..5abb81c 100644
--- a/sites/all/modules/contrib/legal/legal.module
+++ b/sites/all/modules/contrib/legal/legal.module
@@ -423,7 +423,9 @@ function legal_user_login(&$edit, $account) {
   $query = NULL;
   // Deal with destination from password reset one time login link.
   if (arg(0) == 'user' && arg(1) == 'reset') {
-    $query = array('destination' => "user/$uid/edit");
+    $token = drupal_hash_base64(drupal_random_bytes(55));
+    $_SESSION['pass_reset_' . $uid] = $token;
+    $query = array('destination' => "user/$uid/edit?pass-reset-token=" . $token);
   }
 
   if (!empty($_REQUEST['destination'])) {
