diff --git a/entityreference.install b/entityreference.install
index 4e248ef..327cf8e 100644
--- a/entityreference.install
+++ b/entityreference.install
@@ -41,6 +41,7 @@ function entityreference_field_schema($field) {
     }
 
     // Invoke the behaviors to allow them to change the schema.
+    require_once(dirname(__FILE__) . '/entityreference.module');
     foreach (entityreference_get_behavior_handlers($field) as $handler) {
       $handler->schema_alter($schema, $field);
     }
@@ -161,4 +162,4 @@ function entityreference_update_7002() {
       'not null' => TRUE,
     ));
   }
-}
\ No newline at end of file
+}
