diff --git a/chosen.js b/chosen.js
index 427b39d..c8b4a7f 100644
--- a/chosen.js
+++ b/chosen.js
@@ -8,6 +8,7 @@
       options.placeholder_text_multiple = Drupal.settings.chosen.placeholder_text_multiple;
       options.placeholder_text_single = Drupal.settings.chosen.placeholder_text_single;
       options.no_results_text = Drupal.settings.chosen.no_results_text;
+      options.inherit_select_classes = true;
 
       $(Drupal.settings.chosen.selector, context)
         .not('#field-ui-field-overview-form select, #field-ui-display-overview-form select') //disable chosen on field ui
diff --git a/css/chosen-drupal.css b/css/chosen-drupal.css
index c380961..49f5404 100644
--- a/css/chosen-drupal.css
+++ b/css/chosen-drupal.css
@@ -4,4 +4,8 @@
  */
 .chzn-container {
   display: table !important;
+}
+
+.chzn-container.error .chzn-choices {
+  border: 2px solid red;
 }
\ No newline at end of file
