diff --git a/workbench_moderation.module b/workbench_moderation.module
index b652b6f..a21ac67 100644
--- a/workbench_moderation.module
+++ b/workbench_moderation.module
@@ -1100,12 +1100,17 @@ function workbench_moderation_node_data($node) {
     $published = $current;
   }
   else {
-    // Fetch the most recent published revision.
+    // Fetch the most recent published revision as marked in
+    // workbench_moderation.
+    // In some cases node.vid will not be the correct vid we assume to be/get
+    // published (esp. in workbench_moderation_store()), so we rely on the
+    // states workbench_moderation provides.
     $query = db_select('node', 'n');
-    $query->addJoin('INNER', 'node_revision', 'r', 'n.vid = r.vid');
+    $query->addJoin('INNER', 'node_revision', 'r', 'n.nid = r.nid');
     $query->addJoin('LEFT OUTER', 'workbench_moderation_node_history', 'm', 'r.vid = m.vid');
     $query->condition('n.nid', $node->nid)
       ->condition('n.status', 1)
+      ->condition('m.published', 1)
       ->orderBy('m.hid', 'DESC')
       ->fields('r', array('nid', 'vid', 'title', 'timestamp'))
       ->fields('m');
