diff --git a/docroot/modules/file/file.module b/docroot/modules/file/file.module
index 2a8e6c3..de253f0 100644
--- a/docroot/modules/file/file.module
+++ b/docroot/modules/file/file.module
@@ -361,6 +361,7 @@ function file_managed_file_process($element, &$form_state, $form) {
   // Append the '-upload' to the #id so the field label's 'for' attribute
   // corresponds with the file element.
   $element['#id'] .= '-upload';
+  $fid = isset($element['#value']['fid']) ? $element['#value']['fid'] : 0;
 
   // Set some default element properties.
   $element['#progress_indicator'] = empty($element['#progress_indicator']) ? 'none' : $element['#progress_indicator'];
