diff --git a/sites/all/modules/contrib/legal/legal.module b/sites/all/modules/contrib/legal/legal.module
index ee77bbb..912273c 100644
--- a/sites/all/modules/contrib/legal/legal.module
+++ b/sites/all/modules/contrib/legal/legal.module
@@ -639,7 +639,7 @@ function theme_legal_login($variables) {
   $form = $variables['form'];
   $form = theme('legal_display', array('form' => $form));
 
-  $output = '<p>' . t('To continue to use this site please read the Terms & Conditions below, and complete the form to confirm your acceptance.') . '</p>'
+  $output = '<p>' . t('To continue to use this site please read the Terms of Service below, and complete the form to confirm your acceptance.') . '</p>'
 
   if (isset($form['changes']['#value'])) {
     foreach (element_children($form['changes']) as $key) {
