diff --git a/docroot/sites/all/modules/contrib/rate/rate.js b/docroot/sites/all/modules/contrib/rate/rate.js
index 41ea0b4..8093ed3 100644
--- a/docroot/sites/all/modules/contrib/rate/rate.js
+++ b/docroot/sites/all/modules/contrib/rate/rate.js
@@ -28,6 +28,20 @@
     // Random number to prevent caching, see http://drupal.org/node/1042216#comment-4046618
     var random = Math.floor(Math.random() * 99999);
 
+    // convert stacked setting values back into their original strings
+    if (typeof Drupal.settings.rate.basePath != 'string') {
+      for (var key in Drupal.settings.rate) {
+        // skip properties defined in the prototype
+        if (!Drupal.settings.rate.hasOwnProperty(key)) {
+          continue;
+        }
+
+        if (typeof Drupal.settings.rate[key] == 'object') {
+          Drupal.settings.rate[key] = Drupal.settings.rate[key][0];
+        }
+      }
+    }
+
     var q = (Drupal.settings.rate.basePath.match(/\?/) ? '&' : '?') + 'widget_id=' + data.widget_id + '&content_type=' + data.content_type + '&content_id=' + data.content_id + '&widget_mode=' + data.widget_mode + '&token=' + token + '&destination=' + encodeURIComponent(Drupal.settings.rate.destination) + '&r=' + random;
     if (data.value) {
       q = q + '&value=' + data.value;
